src/components/[guild]/activity/ActivityLogFiltersBar/components/FilterTag.tsx

```diff
@@ -0,0 +1,397 @@
+import { ACTION } from "../../constants"
+import { filterNames, useActivityLogFilters } from "./ActivityLogFiltersContext"
+import Dropdown from "./Dropdown"
+import Suggestion from "./Suggestion"
+
+type Props = {
+  filterId: string
+  focusFiltersInput: () => void
+  positionerStyle: Record<string, string>
+} & TagProps
+
+const FilterTag = ({
+  filterId,
+  focusFiltersInput,
+  positionerStyle,
+  ...tagProps
+}: PropsWithChildren<Props>): JSX.Element => {
+  const tagFontColor = useColorModeValue("black", undefined)
+  const closeBtnFocusBgColor = useColorModeValue("blackAlpha.300", "whiteAlpha.500")
+
+  const {
+    getFilter,
+    updateFilter,
+    removeFilter,
+    getFilteredRoleSuggestions,
+    getFilteredRewardSuggestions,
+    getFilteredActionSuggestions,
+  } = useActivityLogFilters()
+  const { id, filter, value } = getFilter(filterId)
+
+  const { roles, guildPlatforms } = useGuild()
+
+  const [isLoading, setIsLoading] = useState(false)
+  const [shouldRemove, setShouldRemove] = useState(value.length === 0)
+
+  const [state, send] = useMachine(
+    combobox.machine({
+      id: `tag-${id}-combobox`,
+      inputBehavior: "autohighlight",
+      onSelect({ value: selectedValue }) {
+        updateFilter({
+          id,
+          filter,
+          value: selectedValue?.trim() ?? "",
+        })
+        focusFiltersInput()
+      },
+      openOnClick: true,
+      autoFocus: true,
+      loop: true,
+    })
+  )
+
+  const {
+    rootProps,
+    controlProps,
+    inputProps,
+    positionerProps,
+    contentProps,
+    getOptionProps,
+    focusedOption,
+    inputValue,
+    setInputValue,
+    focus,
+    isFocused,
+  } = combobox.connect(state, send, normalizeProps)
+
+  useEffect(() => {
+    if (!isFocused) return
+    // Opening the suggestions dropdown
+    send({ type: "CLICK_INPUT" })
+  }, [isFocused])
+
+  const { size, ...filteredInputProps } = inputProps
+
+  const onChange = async (e) => {
+    const newValue = e.target.value
+
+    if (filter !== "userId") {
+      setInputValue(newValue)
+      return
+    }
+
+    const trimmedValue = newValue.trim()
+    setInputValue(trimmedValue)
+
+    if (ADDRESS_REGEX.test(trimmedValue)) {
+      setIsLoading(true)
+      try {
+        const guildUser = await fetcher(`/v2/users/${trimmedValue}`)
+        setInputValue(guildUser.id.toString())
+      } catch {
+        // We just leave the original content of the input
+      } finally {
+        setIsLoading(false)
+      }
+    }
+  }
+
+  return (
+    <>
+      <Tag
+        color={tagFontColor}
+        position="relative"
+        borderColor="transparent"
+        onClick={() => focus()}
+        pr={0}
+        {...tagProps}
+        {...rootProps}
+      >
+        <HStack {...controlProps}>
+          <Text as="span" fontSize="sm" fontWeight="bold">
+            {filterNames[filter]}:
+          </Text>
+
+          {value ? (
+            <>
+              {(() => {
```

Comment:
We could extract this logic to its own component, but I felt like it's not that bad this way. Let me know what's your opinion on this!